### ISSUE TITLE
Accessing compile-time generated service metadata at listener init phase

### ISSUE THREAD

**Issue by sameerajayasoma** [ballerina-platform/ballerina-library#2715]
Service declarations create service objects and attach themselves to listeners by passing references to service objects. Listeners use service objects to extract the information required to route requests to related resources or remote functions. 

Some of these protocol listeners need to access additional compile-time generated service metadata. E.g., GraphQL schema, OpenAPI description, gRPC protobuf mappings. We need to figure out a way to pass such service metadata to listeners. 

The gRPC uses a service-level annotation, but it is generated by the gRPC tool (not compile-time). 

We could use the same approach in other scenarios. What if we generate the OpenAPI description during the compile-time and add a service-level annotation to each HTTP service? We need the `CodeModifier` compiler plugin to implement this, though. 

 Regardless of the implementation complexity, what do you think about this? @shafreenAnfar @daneshk @azinneera

**Comment by shafreenAnfar** [comment]
Yeah, sounds good. If we can annotate the meta information as part of the service, that would be the ideal solution for us.

**Comment by ThisaruGuruge** [comment]
Removing GraphQL label since this is now implemented
